[Chua Chen Ler] iP

lerxcl opened the pull request.

Commit:
Add Gradle support

lerxcl committed:
Complete Level 1, use Scanner to take in inputs

lerxcl committed:
-Modify "Welcome" and "bye" message
-Change name from "Duke" to "Bob"
-Complete Level-2

lerxcl committed:
-Create Task class
-Change list array from String type to Task type
-Complete Level 3

lerxcl committed:
-Create Deadline, Event and Todo classes
-Complete Level 4

lerxcl committed:
Test program with some sample input, edit runtest.bat to properly compile the Java files

lerxcl committed:
-Change from Array to ArrayList data structure
-Implement multiple try-catch blocks for different exceptions encountered so far

lerxcl committed:
-Implement delete feature
-Try-catch potential exceptions

lerxcl committed:
Create ListOfItems class to streamline code in Duke class

lerxcl committed:
Create DukeException class to create customise exceptions

lerxcl committed:
Did more comprehensive testing, which includes all potential wrong commands

lerxcl committed:
Create HandleFile class to handle all code related to processing of stored.txt file

lerxcl committed:
Change all Strings involving date/time to the respective LocalDate and LocalTime class

Deadline now follows this pattern: deadline <description> /<connector word> DD/MM/YYYY HHmm(optional)

Event now follows this pattern:
event <description> /<connector word> DD/MM/YYYY HHmm-HHmm

lerxcl committed:
Change DateTimeFormatter pattern from "d/MM/yyyy" to "d/M/yyyy" to accept single digit month input

lerxcl committed:
Implement ability to find deadlines due on a specific date

command: show items due by <DD/MM/YYYY>

lerxcl committed:
Implement feature to find deadlines/events before a specific date + time

command: items due before <DD/MM/YYYY> <HHmm> (optional)

Change command of "... due by" to:
items due by <DD/MM/YYYY>

Reason for change in command is to omit the "show" for simplicity

lerxcl committed:
Fix bug of index of substring being incorrect for checking items before/by...

Fix bug of printing out message when an event is not created properly

lerxcl committed:
Fix logic of "items due before..."

Use !isAfter instead of isBefore method for LocalDate and LocalTime class

Fix bug of printing out message when an todo/deadline is not created properly

lerxcl committed:
Merge branch 'branch-Level-7' into master

lerxcl committed:
Resolve merge conflicts between master branch and Level-8 branch

lerxcl committed:
Resolve bugs after merging Level-7 and 8

Modify the constructors of Deadline, Event to handle file inputs since now date and time are in LocalDate and LocalTime format instead of Strings

lerxcl committed:
Create Parser and Ui class to make the code more OOP

HandleFile class = "Storage"
ListOfItems class = "TaskList"

lerxcl committed:
Implement JUnit 5 for testing

Create 5 test cases:
1. testTodo
2. testDeadlineDate
3. testDeadlineDateTime
4. testEvent
5. testDone

lerxcl committed:
Add DateTimeParseException for addItem

Tweak DukeException messages

lerxcl committed:
Commit to add A-Jar tag

lerxcl committed:
Add JavaDoc to every class and public method

lerxcl committed:
Tweak code to adhere to coding standard

lerxcl committed:
Add find method in ListOfItems, and "find" keyword in Parser

lerxcl committed:
Merge tag 'A-JavaDoc' into master

A-JavaDoc

lerxcl committed:
Resolve conflicts between A-JavaDoc and A-CodingStandard branch

lerxcl committed:
Merge branch 'branch-Level-9' into master

lerxcl committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

lerxcl committed:
Add Gradle support

Test:
- Clean project
- Running checkstyleMain and checkstyleTest
- Running JUnit tests (modify to wipe "stored.txt" everytime the test starts)
- Creating JAR files with shadowJar

lerxcl committed:
Merge branch 'branch-A-Gradle' into master

lerxcl committed:
Ensure checkstyleMain and checkstyleTest (Gradle) clears the code with no errors

lerxcl committed:
Implement JavaFX for GUI

Create:
- DialogBox class
- Launcher class
- Main class
- MainWindow class
- DialogBox.fxml and MainWindow.fxml
- 2 images (Bob.jpg and User.png)

Edit:
- build.gradle
- Most of ListOfItems methods change from void to String type (for it to return a Label), thus there is a need to modify the constructors for Parser, Ui and ListOfItems class as HandleFile can no longer run its command after returning the label
- Duke: add getResponse(), change arguments for initialising the variables

lerxcl committed:
Add padding between profile picture and text

Add application name: "Bob the Chatbot"

lerxcl committed:
Update Javadocs after implementing JavaFX

Many of the methods used to be of type void, but now it needs to return a String type, thus there is a need to add "@return ..." for these modified methods.

Remove full-stop in every "@param ..." to follow convention.

lerxcl committed:
Correct DialogBox size when list gets too long

esmanda3w commented on src/main/java/HandleFile.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    /**
+     * It checks whether if "stored.txt" file exists, which it will pass to readFile().
```
Perhaps could start the sentence with a verb in the present tense to abide by the coding standard? (eg Checks whether...)

esmanda3w commented on src/main/java/ListOfItems.java:
```diff
@@ -0,0 +1,353 @@
+     *
+     * @param input a line from "stored.txt".
+     */
+    protected void addStored(String input) {
```
Perhaps could use a more descriptive name - what is stored? Add stored list?

esmanda3w commented on src/test/java/DukeTest.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Test
+    void testTodo() {
```
Maybe using the three part format for naming test methods can provide more clarity on what you are testing.

esmanda3w commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,38 @@
+    /**
+     *  Marks a task as complete.
+     */
+    public void markedDone() {
```
Perhaps could change the verb into present tense, such as markAsDone?

esmanda3w commented on src/main/java/ListOfItems.java:
```diff
@@ -0,0 +1,353 @@
+     * @param input user input.
+     * @throws DukeException if incomplete commands are given.
+     */
+    protected void addItem(String input) throws DukeException {
```
This method seems to be doing too much work, perhaps can consider abstracting the parsing into the Parser class? I noticed this issue in several other places.

esmanda3w commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,58 @@
+            //stored data
+            DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("d/MMM/yyyy");
+            DateTimeFormatter timeFormat = DateTimeFormatter.ofPattern("h:mma");
+            String[] dateTime = duration.split(" ");
```
Perhaps can consider using the plural form to distinguish between multivalued variables.

esmanda3w commented on src/main/java/ListOfItems.java:
```diff
@@ -0,0 +1,353 @@
+/**
+ * Contains the task list, handles all of the list manipulations + checks.
+ */
+public class ListOfItems {
```
Perhaps it would be better to name the class as ItemList?

esmanda3w commented on src/main/java/ListOfItems.java:
```diff
@@ -0,0 +1,353 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
```
Would it be better to separate the similar imports into different blocks?

theodoreleebrant replied:
Agreed, it would probably be better if the `java.util.ArrayList` is separated in a new line.

esmanda3w reviewed (commented):
Generally the code seems alright for the most part, but would require a little tidying to follow the coding standard, as well as abstraction of some logic to prevent methods from doing too much work. :)

theodoreleebrant commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+ * Deadline is a type of Task, which takes in a specific date/time.
+ */
+public class Deadline extends Task {
+
```
Could you remove the line break here, as per Google Java Style Guide?

theodoreleebrant commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,35 @@
+ * @author Chen Ler
+ */
 public class Duke {
+
```
Could you remove the line break here, as per Google Java Style Guide?

theodoreleebrant commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,58 @@
+ * Event is a type of Task, which takes in a specific date/time.
+ */
+public class Event extends Task {
+
```
Could you remove the line break here, as per Google Java Style Guide?

theodoreleebrant commented on src/main/java/HandleFile.java:
```diff
@@ -0,0 +1,80 @@
+ * Handles all of the file operations of "stored.txt" file.
+ */
+public class HandleFile {
+
```
Could you remove the line break here, as per Google Java Style Guide?

theodoreleebrant commented on src/main/java/ListOfItems.java:
```diff
@@ -0,0 +1,353 @@
+ */
+public class ListOfItems {
+
+    protected List<Task> list;
```
Perhaps a more descriptive variable name would be more apt here.

theodoreleebrant reviewed (commented):
LGTM, just a few nits to fix.

lerxcl committed:
Add icon to application

lerxcl committed:
Add "dark" mode

Make profile pictures circular

lerxcl committed:
Add background color to inputs + outputs

lerxcl committed:
Merge branch 'branch-Level-10' into master

lerxcl committed:
Add assertions for checking item id > 0

lerxcl committed:
Add more assertions

-For checking item type is correct after adding to list
-Item is marked as done after using "done" command

lerxcl committed:
Correct some code standard violation mention by peer reviews

-Remove spacing in-between class declaration and variables
-Edit HandFile's Javadoc to start with a verb "Checks..." instead of starting with an "It..."
-Change addStored() naming to addStoredList()
-Change markedDone() naming to markAsDone()
-Segregate different imports into blocks in ListOfItems

lerxcl committed:
-Use else if (type == 'E') instead of else to prevent assumptions.
-Branch out addItem() into the respective task types to reduce method size.
-Add JavaDocs for addTodo(), addDeadline() and addEvent()
-Create separate booleans in checkBy and checkBefore for better code readability.

lerxcl committed:
Merge pull request #1 from lerxcl/branch-A-Assertions

Branch A-Assertions

lerxcl committed:
Resolve merge conflicts between A-Assertions and A-CodeQuality

lerxcl committed:
Merge pull request #2 from lerxcl/branch-A-CodeQuality

Branch A-CodeQuality

lerxcl committed:
Remove extra spacing in ListOfItems

lerxcl committed:
Add a Help class to give a list of commands a user can enter

-Modify Parser class to detect "help" keyword
-Modify welcome message in Ui class to inform the user of a "help" function

lerxcl committed:
Merge branch 'branch-C-Help' into master

lerxcl committed:
Tweak GUI to have more padding between Bob and user

-Minor edit to welcome message
-Tweak layout of help message so that it is easier to see

lerxcl committed:
Upload screenshot and create README

lerxcl committed:
Modify "bye" command behaviour

-Ui class calls stage.close() to close the GUI window
